AutoRouteManager.php

```diff
@@ -64,10 +71,30 @@ public function __construct(
      */
     public function buildUriContextCollection(UriContextCollection $uriContextCollection)
     {
-        $this->getUriContextsForDocument($uriContextCollection);
+        $this->collectionBuilder->build($uriContextCollection);
 
         foreach ($uriContextCollection->getUriContexts() as $uriContext) {
-            $existingRoute = $this->adapter->findRouteForUri($uriContext->getUri(), $uriContext);
+            $subjectObject = $uriContextCollection->getSubjectObject();
+
+            if (null !== $uriContext->getLocale()) {
+                $subjectObject = $this->adapter->translateObject($uriContextCollection->getSubjectObject(), $uriContext->getLocale());
+            }
+
+            if (null === $subjectObject) {
+                @trigger_error('AdapterInterface::translateObject() has to return the subjectObject as of version 1.1, support for by reference will be removed in 2.0.', E_USER_DEPRECATED);
+            }
+
+            if ($subjectObject && $subjectObject !== $uriContext->getSubjectObject()) {
+                $subjectObject = $uriContext->newSubjectInstance($subjectObject);
```

Comment:
This does not seem right - it should assign a new instance of the `UriContext` (with the new subject) to the `$uriContext` variable.